include/alpaka/dev/DevOmp5.hpp

```diff
@@ -0,0 +1,365 @@
+                    //-----------------------------------------------------------------------------
+                    DevOmp5Impl(DevOmp5Impl const &) = delete;
+                    //-----------------------------------------------------------------------------
+                    DevOmp5Impl(DevOmp5Impl &&) = delete;
+                    //-----------------------------------------------------------------------------
+                    auto operator=(DevOmp5Impl const &) -> DevOmp5Impl & = delete;
+                    //-----------------------------------------------------------------------------
+                    auto operator=(DevOmp5Impl &&) -> DevOmp5Impl & = delete;
+                    //-----------------------------------------------------------------------------
+                    ~DevOmp5Impl() = default;
+
+                    //-----------------------------------------------------------------------------
+                    ALPAKA_FN_HOST auto getAllExistingQueues() const
+                    -> std::vector<std::shared_ptr<queue::IGenericThreadsQueue<DevOmp5>>>
+                    {
+                        std::vector<std::shared_ptr<queue::IGenericThreadsQueue<DevOmp5>>> vspQueues;
+
+                        std::lock_guard<std::mutex> lk(m_Mutex);
+                        vspQueues.reserve(m_queues.size());
+
+                        for(auto it = m_queues.begin(); it != m_queues.end();)
+                        {
+                            auto spQueue(it->lock());
+                            if(spQueue)
+                            {
+                                vspQueues.emplace_back(std::move(spQueue));
+                                ++it;
+                            }
+                            else
+                            {
+                                it = m_queues.erase(it);
+                            }
+                        }
+                        return vspQueues;
+                    }
+
+                    //-----------------------------------------------------------------------------
+                    //! Registers the given queue on this device.
+                    //! NOTE: Every queue has to be registered for correct functionality of device wait operations!
+                    ALPAKA_FN_HOST auto registerQueue(std::shared_ptr<queue::IGenericThreadsQueue<DevOmp5>> spQueue)
+                    -> void
+                    {
+                        std::lock_guard<std::mutex> lk(m_Mutex);
+
+                        // Register this queue on the device.
+                        m_queues.push_back(spQueue);
+                    }
+
+                    int iDevice() const {return m_iDevice;}
+
+                private:
+                    std::mutex mutable m_Mutex;
+                    std::vector<std::weak_ptr<queue::IGenericThreadsQueue<DevOmp5>>> mutable m_queues;
+                    int m_iDevice = 0;
```

Comment:
Can we make this `const` and remove the `=0`? It is always initialized in the constructor.